.github/pull_request_template.release.md

```diff
@@ -11,6 +11,7 @@ Target: upstream/develop (always!)
 - [ ] Tag the develop branch with ``vX.Y.Z-rc.N`` where N starts with 1, and increment for every time you return to this step due to new pull requests.
 - [ ] Run performance regression tests on Summit, Crusher/Frontier, and an additional machine with debug assertions enabled (e.g., Wildstyle).
 - [ ] Update documentation with release notes from all pull requests newly included in the release.
+- [ ] Check for (and delete if found) code marked as "deprecated: to be removed in vX.Y".
```

Comment:
This can be helped by marking the deprecated features by a macro that turns into a Warning/Reminder/Error when the version number reaches the target version number.

Reply:
Good idea... the only potential issue is that we would be unable to tag the develop branch upstream as `next.rc-1` because then the tag would cause the deprecations to trigger.